realm/realm-library/src/objectServer/java/io/realm/mongodb/mongo/iterable/AggregateIterable.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.mongodb.mongo.iterable;
+
+import java.util.Collection;
+import java.util.List;
+
+import io.realm.internal.common.TaskDispatcher;
+import io.realm.internal.objectstore.OsAggregateIterable;
+import io.realm.mongodb.mongo.MongoCollection;
+
+/**
+ * Specific iterable for {@link MongoCollection#aggregate(List)} operations.
+ *
+ * @param <ResultT> The type to which this iterable will decode documents.
+ */
+public class AggregateIterable<ResultT> extends MongoIterable<ResultT> {
+
+    private OsAggregateIterable<ResultT> osAggregateIterable;
+
+    public AggregateIterable(final TaskDispatcher dispatcher,
+                             final OsAggregateIterable<ResultT> osAggregateIterable) {
+        super(dispatcher);
+        this.osAggregateIterable = osAggregateIterable;
+    }
+
+    @Override
```

Comment:
Missing Javadoc

Reply:
Becomes a private method after removing the unnecessary OS classes.